emulator.c

```diff
@@ -269,6 +289,10 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       }
     case 0xc1: // NOLINT
       printf("POP B");
+      cpu->c = cpu->memory[cpu->sp];
```

Comment:
I would use the function cpu_read_mem rather than directly accessing memory. Brings it in line with the rest of the opcodes and if we decide to change the way the read_mem function works down the road it'll change it here too.